src/code-util.jl

```diff
@@ -281,13 +302,23 @@ function extract_ics(results)
         return ics
     end
 
-    # Diagnostic categories have identical ordering so we can use row to find matches
-    # across categories, starting with row 1
     for i in 1:length(results[IC_LINE_KEY][1])
         line = extract_detail(results, IC_LINE_KEY, 2, i)
         report = extract_detail(results, IC_REPORT_KEY, 2, i)
 
-        ic = (; line, report)
+        # IC Diagnostic values are indexed by hash and type so we extract them separately
+        h = extract_detail(results, IC_LINE_KEY, 1, i)
+        vs = extract_ic_results(results, IC_OUTPUT_KEY, 2, h)
```

Comment:
there can be any arity of values, does this account for that?

Reply:
Good point! It does not account for that, yet.

Reply:
Added handling of multiple arities, and tests.